<details><summary>Failed to submit graph for a5ebdf1b6dbb8b93df80fe13c69f142b4fb9b079
</summary>

Message:
Failed to run command: 'pbp generate from-feedstock --output /tmp/tmpw61kju6_/lumen-feedstock.yaml --custom-metadata-file /tmp/tmpw61kju6_/custom_metadata.yaml --release-name lumen-feedstock-8-a5ebdf1b6dbb8b93df80fe13c69f142b4fb9b079 lumen-feedstock/recipe', with error: 'INFO:root:Found feedstock lumen-feedstock at /tmp/tmpw61kju6_/lumen-feedstock/recipe
INFO:root:Build/Test graph from lumen-feedstock
Created by sbx_user1051 on 2026-07-07 10:46:09
Command line: /opt/conda/envs/webhook-listener/bin/pbp generate from-feedstock --output /tmp/tmpw61kju6_/lumen-feedstock.yaml --custom-metadata-file /tmp/tmpw61kju6_/custom_metadata.yaml --release-name lumen-feedstock-8-a5ebdf1b6dbb8b93df80fe13c69f142b4fb9b079 lumen-feedstock/recipe

INFO:root:Creating graph from feedstock at: lumen-feedstock
INFO:pbpcli.utils.common:Setting env var: CONDA_SUBDIR = linux-64.
INFO:pbpcli.utils.expand_variants:Elapsed time for get_package_variants: 0.11 seconds
INFO:pbpcli.utils.expand_variants:Matrix vars: 4 unchanging vars:(467), changing vars (4): ['blas_impl', 'cuda_compiler_version', 'numpy', 'python']
INFO:pbpcli.utils.expand_variants:Elapsed time for preparation for render: 0.00 seconds
INFO:pbpcli.utils.common:Setting env var: CONDA_PKGS_DIRS = /tmp/conda_pkgs_7ypi5r35.
INFO:pbpcli.utils.common:Unsetting env var: CONDA_PKGS_DIRS
INFO:pbpcli.utils.common:Unsetting env var: CONDA_SUBDIR
Traceback (most recent call last):
  File "/opt/conda/envs/webhook-listener/bin/pbp", line 10, in <module>
    sys.exit(cli())
             ~~~^^
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/click/core.py", line 1524, in __call__
    return self.main(*args, **kwargs)
           ~~~~~~~~~^^^^^^^^^^^^^^^^^
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/click/core.py", line 1445, in main
    rv = self.invoke(ctx)
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/click/core.py", line 1912, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
                           ~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/click/core.py", line 1912, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
                           ~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/click/core.py", line 1308, in invoke
    return ctx.invoke(self.callback, **ctx.params)
           ~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/click/core.py", line 877, in invoke
    return callback(*args, **kwargs)
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/pbpcli/cli/generate.py", line 321, in wrapper
    return func(*args, **kwargs)
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/pbpcli/cli/generate.py", line 591, in from_feedstock
    build_configs, test_configs = expand_build_matrix(
                                  ~~~~~~~~~~~~~~~~~~~^
        feedstock.recipe_dir_path, src_channels, parsed_user_variants
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    )
    ^
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/pbpcli/cli/generate.py", line 97, in expand_build_matrix
    _targets, build_configs, test_configs = variants_for_feedstock_and_platform(
                                            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
        recipe_dir_path, platform, src_channels, user_variants
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    )
    ^
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/pbpcli/utils/expand_variants.py", line 577, in variants_for_feedstock_and_platform
    metadata_tuples = api.render(
        parsed.recipe,
    ...<2 lines>...
        variants=user_variants,
    )
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/conda_build/api.py", line 68, in render
    return render_metadata_tuples(
        metadata_tuples,
    ...<3 lines>...
        bypass_env_check=bypass_env_check,
    )
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/conda_build/render.py", line 1097, in render_metadata_tuples
    for od, om in meta.get_output_metadata_set(
                  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
        permit_unsatisfiable_variants=permit_unsatisfiable_variants,
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        permit_undefined_jinja=not finalize,
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        bypass_env_check=bypass_env_check,
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    ):
    ^
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/conda_build/metadata.py", line 2802, in get_output_metadata_set
    _check_run_constrained(output_tuples)
    ~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/conda_build/metadata.py", line 521, in _check_run_constrained
    raise RecipeError("\n".join(["", *errors]))
conda_build.exceptions.RecipeError: 
- Output 'lumen' has invalid run_constrained item: llama-cpp-python >=0.3.0 - pygwalker >=0.4.9.11. Reason: Invalid spec 'llama-cpp-python >=0.3.0 - pygwalker >=0.4.9.11': Invalid version '0.3.0-pygwalker>=0.4.9.11': invalid character(s)
- Output 'lumen' has invalid run_constrained item: llama-cpp-python >=0.3.0 - pygwalker >=0.4.9.11. Reason: Invalid spec 'llama-cpp-python >=0.3.0 - pygwalker >=0.4.9.11': Invalid version '0.3.0-pygwalker>=0.4.9.11': invalid character(s)
- Output 'lumen' has invalid run_constrained item: llama-cpp-python >=0.3.0 - pygwalker >=0.4.9.11. Reason: Invalid spec 'llama-cpp-python >=0.3.0 - pygwalker >=0.4.9.11': Invalid version '0.3.0-pygwalker>=0.4.9.11': invalid character(s)
- Output 'lumen' has invalid run_constrained item: llama-cpp-python >=0.3.0 - pygwalker >=0.4.9.11. Reason: Invalid spec 'llama-cpp-python >=0.3.0 - pygwalker >=0.4.9.11': Invalid version '0.3.0-pygwalker>=0.4.9.11': invalid character(s)
'
aws lambda request id: d72b2b50-9aac-5ac7-87d9-49964182aefb
[View in Grafana](https://grafana.anacondaconnect.com/explore?schemaVersion=1&panes=%7B%2286g%22:%7B%22datasource%22:%22loki-central%22,%22queries%22:%5B%7B%22refId%22:%22A%22,%22expr%22:%22%7B__aws_cloudwatch_log_group%3D%5C%22%2Faws%2Flambda%2Fpbp-graph-submitter-prod%5C%22%7D%20%7C%20json%20%7C%20requestId%20%3D%20%60d72b2b50-9aac-5ac7-87d9-49964182aefb%60%22,%22queryType%22:%22range%22,%22datasource%22:%7B%22type%22:%22loki%22,%22uid%22:%22loki-central%22%7D,%22editorMode%22:%22builder%22,%22direction%22:%22backward%22%7D%5D,%22range%22:%7B%22from%22:%22now-3h%22,%22to%22:%22now%22%7D,%22panelsState%22:%7B%22logs%22:%7B%22columns%22:%7B%220%22:%22Time%22,%221%22:%22Line%22%7D,%22visualisationType%22:%22table%22,%22labelFieldName%22:%22labels%22%7D%7D%7D%7D&orgId=1)

</details>

--- a/recipe/meta.yaml
+++ b/recipe/meta.yaml
@@ -7,8 +7,10 @@
 # - lumen-ai-local: Metapackage that adds huggingface_hub for local AI models
 # - lumen-ai-openai: Metapackage that adds OpenAI support
 # - lumen-ai-mistralai: Metapackage that adds MistralAI support
+# - lumen-ai-anthropic: Metapackage that adds Anthropic support
+
 package:
-  name: {{ name|lower }}-bundle
+  name: {{ name }}-bundle
   version: {{ version }}
 
 source:
@@ -19,11 +21,12 @@ build:
   number: 0
 
 outputs:
-  - name: {{ name|lower }}
+  # Main lumen package
+  - name: {{ name }}
     build:
       entry_points:
         - lumen = lumen.command:main
-      skip: true  # [py<310]
+      skip: true  # [py<311]
       script: python -m pip install . --no-deps --no-build-isolation --ignore-installed --no-cache-dir -vv
     requirements:
       host:
@@ -33,23 +36,38 @@ outputs:
         - hatch-vcs
       run:
         - python
-        - panel >=1.6.3
-        - hvplot
-        - holoviews >=1.17.0
-        - pandas
         - bokeh
+        - chardet
+        - duckdb >=1.2.0
+        - gw-dsl-parser
+        - lxml
+        - griffe
+        - holoviews >=1.17.0
+        - hvplot
+        - instructor >=1.14.3
         - intake <2
         - jinja2 >3.0
+        - markitdown
+        - nbformat
         - numpy
         - packaging
-        - param >=1.9.0
-        - sqlglot
-      run_constrained:
-        - instructor >=1.6.4
-        - pydantic >=2.8.0
+        - pandas
+        - panel >=1.7.5
         - panel-graphic-walker >=0.6.4
+        - panel-material-ui >=0.8.1
+        - panel-splitjs >=0.3.2
+        - param >=2.2.1
+        - pyarrow
+        - pydantic >=2.8.0
+        - pydantic-extra-types
+        - semchunk
+        - sqlglot
+        - tabulate
+        - tiktoken
+        - vl-convert-python
+      run_constrained:
         - llama-cpp-python >=0.3.0
-        - duckdb >=1.2.0
+        - pygwalker >=0.4.9.11
     test:
       source_files:
         - lumen/tests
@@ -65,7 +83,6 @@ outputs:
         - pip
         - pydantic
         - duckdb
-        - instructor
         - griffe
         - pydantic-extra-types
         - jsonschema
@@ -91,6 +108,105 @@ outputs:
       doc_url: https://lumen.holoviz.org/
       dev_url: https://github.com/holoviz/lumen
 
+  # Lumen with AI capabilities
+  - name: {{ name }}-ai
+    build:
+      skip: true  # [py<311 or win]
+    requirements:
+      host:
+        - python
+        - pip
+      run:
+        - python
+        - {{ pin_subpackage('lumen', exact=True) }}
+        - griffe
+        - nbformat
+        - duckdb >=1.2.0
+        - pyarrow
+        - instructor >=1.14.3
+        - pydantic >=2.8.0
+        - pydantic-extra-types
+        - panel-graphic-walker >=0.6.4
+        - pygwalker >=0.4.9.11
+        - markitdown
+        - tiktoken
+        - chardet
+        - semchunk
+    about:
+      home: https://github.com/holoviz/lumen
+      summary: Lumen with AI capabilities
+      license: BSD-3-Clause
+      license_family: BSD
+
+  # Lumen with huggingface_hub for local AI models
+  - name: {{ name|lower }}-ai-local
+    build:
+      skip: true  # [py<311 or win]
+    requirements:
+      host:
+        - python
+      run:
+        - python
+        - {{ pin_subpackage('lumen-ai', exact=True) }}
+        - huggingface_hub
+        - hf-xet
+    about:
+      home: https://github.com/holoviz/lumen
+      summary: Lumen with huggingface_hub for local AI models
+      license: BSD-3-Clause
+      license_family: BSD
+
+  # Lumen with OpenAI support
+  - name: {{ name|lower }}-ai-openai
+    build:
+      skip: true  # [py<311 or win]
+    requirements:
+      host:
+        - python
+      run:
+        - python
+        - {{ pin_subpackage('lumen-ai', exact=True) }}
+        - openai
+    about:
+      home: https://github.com/holoviz/lumen
+      summary: Lumen with OpenAI support
+      license: BSD-3-Clause
+      license_family: BSD
+
+  # Lumen with MistralAI support
+  - name: {{ name|lower }}-ai-mistralai
+    build:
+      skip: true  # [py<311 or win]
+    requirements:
+      host:
+        - python
+      run:
+        - python
+        - {{ pin_subpackage('lumen-ai', exact=True) }}
+        - mistralai
+    about:
+      home: https://github.com/holoviz/lumen
+      summary: Lumen with MistralAI support
+      license: BSD-3-Clause
+      license_family: BSD
+
+  # Lumen with Anthropic support
+  - name: {{ name|lower }}-ai-anthropic
+    build:
+      skip: true  # [py<311 or win]
+    requirements:
+      host:
+        - python
+      run:
+        - python
+        - {{ pin_subpackage('lumen-ai', exact=True) }}
+        - anthropic
+    about:
+      home: https://github.com/holoviz/lumen
+      summary: Lumen with Anthropic support
+      license: BSD-3-Clause
+      license_family: BSD
+
 about:
   home: https://github.com/holoviz/lumen
   summary: Illuminate your data. A monitoring solution built on Panel
@@ -106,94 +222,3 @@ extra:
   recipe-maintainers:
     - Hoxbro
     - philippjfr
-
-<conda_recipe_manager.types.SentinelType object at 0x7f982191e510>:  # - lumen-ai-anthropic: Metapackage that adds Anthropic support
-  - name: {{ name|lower }}-ai
-    build:
-      # pygwalker not available on win
-      skip: true  # [py<310 or win]
-    requirements:
-      host:
-        - python
-        - pip
-      run:
-        - python
-        - {{ pin_subpackage('lumen', exact=True) }}
-        - griffe
-        - nbformat
-        - duckdb >=1.2.0
-        - pyarrow
-        - instructor >=1.6.4
-        - pydantic >=2.8.0
-        - pydantic-extra-types
-        - panel-graphic-walker >=0.6.4
-        - pygwalker >=0.4.9.11
-        - markitdown
-        - tiktoken
-        - chardet
-        - semchunk
-    about:
-      home: https://github.com/holoviz/lumen
-      summary: Lumen with AI capabilities
-      license: BSD-3-Clause
-      license_family: BSD
-  - name: {{ name|lower }}-ai-local
-    build:
-      skip: true  # [py<310 or win]
-    requirements:
-      host:
-        - python
-      run:
-        - python
-        - {{ pin_subpackage('lumen-ai', exact=True) }}
-        - huggingface_hub
-    about:
-      home: https://github.com/holoviz/lumen
-      summary: Lumen with huggingface_hub for local AI models
-      license: BSD-3-Clause
-      license_family: BSD
-  - name: {{ name|lower }}-ai-openai
-    build:
-      skip: true  # [py<310 or win]
-    requirements:
-      host:
-        - python
-      run:
-        - python
-        - {{ pin_subpackage('lumen-ai', exact=True) }}
-        - openai
-    about:
-      home: https://github.com/holoviz/lumen
-      summary: Lumen with OpenAI support
-      license: BSD-3-Clause
-      license_family: BSD
-  - name: {{ name|lower }}-ai-mistralai
-    build:
-      skip: true  # [py<310 or win]
-    requirements:
-      host:
-        - python
-      run:
-        - python
-        - {{ pin_subpackage('lumen-ai', exact=True) }}
-        - mistralai
-    about:
-      home: https://github.com/holoviz/lumen
-      summary: Lumen with MistralAI support
-      license: BSD-3-Clause
-      license_family: BSD
-  - name: {{ name|lower }}-ai-anthropic
-    build:
-      skip: true  # [py<310 or win]
-    requirements:
-      host:
-        - python
-      run:
-        - python
-        - {{ pin_subpackage('lumen-ai', exact=True) }}
-        - anthropic
-    about:
-      home: https://github.com/holoviz/lumen
-      summary: Lumen with Anthropic support
-      license: BSD-3-Clause
-      license_family: BSD

